contracts/index_executor/src/task.rs

```diff
@@ -113,22 +129,91 @@ impl Task {
         client
             .session()
             .put(b"pending_tasks".as_ref(), pending_tasks.encode());
+        pink_extension::debug!("Task::init_and_submit: exits");
+        Ok(())
+    }
+
+    /// scan for bridge steps and give the first step on each chain a initial block info
+    pub fn setup_bridge_steps(&mut self, context: &Context) -> Result<(), &'static str> {
+        for (_i, step) in self.steps.iter_mut().enumerate() {
+            if let StepMeta::Bridge(bridge_step) = &step.meta {
+                let found = self
+                    .bridges_last_blockinfo
+                    .iter()
+                    .position(|x| x.0 == bridge_step.dest_chain);
+                if found.is_none() {
+                    let chain = context
+                        .graph
+                        .get_chain(bridge_step.dest_chain.clone())
+                        .ok_or("MissingChain")?;
+                    let blockinfo = get_latest_event_block_info(&chain.tx_indexer, &self.recipient)
+                        .or(Err("Can't find timestamp"))?;
+                    pink_extension::debug!(
+                        "setup_bridge_steps: blockinfo {:?} from {}, {}",
+                        blockinfo,
+                        &chain.tx_indexer,
+                        hex::encode(&self.recipient)
+                    );
+                    // we know the Task will be saved on pallet!
+                    self.bridges_last_blockinfo.push((
+                        bridge_step.dest_chain.clone(),
+                        (blockinfo.block_number, blockinfo.index_in_block),
+                    ))
+                }
+            }
+        }
+        pink_extension::debug!(
+            "setup_bridge_steps: successfully exits with {:?}",
+            self.bridges_last_blockinfo
+        );
         Ok(())
     }
 
     // Recover execution status according to on-chain storage
     pub fn sync(&mut self, context: &Context, _client: &SubstrateRollupClient) {
-        for step in self.steps.iter() {
-            // A initialized task must have nonce applied
-            if step.sync_check(step.nonce.unwrap(), context) == Ok(true) {
+        // don't need to setup bridge steps because that info is saved on the init_and_submit process.
+        for i in 0..self.steps.len() {
+            if let Ok((true, extra)) =
+                self.steps[i].sync_check(self.steps[i].nonce.unwrap(), context)
+            {
                 self.execute_index += 1;
                 // If all step executed successfully, set task as `Completed`
                 if self.execute_index as usize == self.steps.len() {
                     self.status = TaskStatus::Completed;
                     break;
                 }
+
+                if let ExtraResult::BlockInfo(block_info) = extra {
+                    // the last step is a bridge step, check its dest chain
```

Comment:
Not quite understand why we need update blockchain info of last bridge step, let's discuss offline